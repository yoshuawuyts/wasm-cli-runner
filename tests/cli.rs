use insta_cmd::{assert_cmd_snapshot, get_cargo_bin};
use std::process::Command;

fn cli() -> Command {
    Command::new(get_cargo_bin("wasm-cli-runner"))
}

#[test]
fn has_a_version() {
    assert_cmd_snapshot!(cli().arg("--version"));
}

#[test]
fn path_is_required() {
    assert_cmd_snapshot!(cli());
}

#[test]
fn path_is_reachable_on_disk() {
    assert_cmd_snapshot!(cli().arg("noop.wasm"));
}

#[test]
fn path_ends_with_wasm() {
    assert_cmd_snapshot!(cli().arg("tests/binaries/time.wasm"));
}

// #[test]
// fn path_is_a_wasm_component() {
//     let mut cmd = Command::cargo_bin("wasm-cli-runner").unwrap();
//     let assert = cmd.write_stdin("noop.wasm").assert();
//     assert.failure().stderr("Could not find file at location\n");
// }
