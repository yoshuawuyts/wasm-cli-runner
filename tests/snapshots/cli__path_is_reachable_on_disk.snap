---
source: tests/cli.rs
info:
  program: wasm-cli-runner
  args:
    - noop.wasm
---
success: false
exit_code: 1
----- stdout -----

----- stderr -----
Error: Os { code: 2, kind: NotFound, message: "No such file or directory" }
